proto/src/determined/api/v1/template.proto

```diff
@@ -51,7 +51,7 @@ message GetTemplateResponse {
 // created
 message PutTemplateRequest {
   // The template to put.
-  determined.template.v1.Template template = 2;
+  determined.template.v1.Template template = 1;
```

Comment:
Isn't this a breaking change?  Or... how does this work?

Reply:
this would be for anyone directly connecting to the grpc server which we don't have any AFAIK so I'm just cleaning it up here. we'll add the breaking change check here https://determinedai.atlassian.net/browse/DET-4209

Reply:
a sneak peak
```
hrz ➜ proto git:(master) ✗ buf check breaking --against-input /tmp/buf.image.bin                                            
src/determined/api/v1/template.proto:52:1:Previously present field "2" with name "template" on message "PutTemplateRequest" was deleted.

```

Reply:
Non-blocking, but it seems like it doesn't buy us anything to make this change, while it is a breaking change.  I'd just as soon leave it alone on principle.  Non-blocking though because it's probably true that nobody is going to hit this.

Reply:
if we think we want to order fields sequentially and that it is common practice and might be more developer-friendly, now would be a good time to order them especially for messages where we have not intentionally reserved fields (https://developers.google.com/protocol-buffers/docs/proto3#assigning_field_numbers). In a lot of our messages, the field numbers are out of order just because they have been heavily under development and we haven't published gprc APIs yet and no other reason. with that said I'm fine with leaving it as is. @stoksc wdy think?

Reply:
as I said, "non-blocking".